providers/common/include/prov/fipsindicator.h

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2023-2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifdef FIPS_MODULE
+
+/*
+ * There may be multiple settables associated with an algorithm that allow
+ * overriding the default status.
+ * We associate an id with each of these.
+ */
+# define OSSL_FIPS_IND_SETTABLE0 0
+# define OSSL_FIPS_IND_SETTABLE1 1
+# define OSSL_FIPS_IND_SETTABLE2 2
+
+# include <openssl/core.h> /* OSSL_CALLBACK, OSSL_LIB_CTX */
+# include <openssl/indicator.h>
+# include "crypto/types.h"
+# include <openssl/ec.h>
+
+#define OSSL_FIPS_IND_MAX_SETTABLE 8
+
+#define OSSL_FIPS_IND_STATE_UNKNOWN    -1  /* Initial unknown state */
+#define OSSL_FIPS_IND_STATE_STRICT      1  /* Strict enforcement */
+#define OSSL_FIPS_IND_STATE_TOLERANT    0  /* Relaxation of rules */
+
+/*
+ * Assume we are approved checks that it does
+ * in different stages that happen before the final output operation.
+ * approved_flags are used to maintain these states as bits. approved_set
+ * is also a bitmask where values of 1 indicate that the approved_flags have been
+ * set.
+ *
+ * If a FIPS approved check fails then we must decide what to do in this case.
+ * In strict mode we would just fail at this point.
+ * To override strict mode we either need to have a settable variable or have a
+ * fips config flag that overrides strict mode.
+ * If strict_checks_set indicates that we have not yet read the strict value.
+ *
+ */
+typedef struct ossl_fips_ind_st {
+    unsigned int approved;
```

Comment:
do I assume right the member `approved` corresponds to `approved_flags` the comment above is talking about?

Reply:
I have removed the flags now - it is just the one field now.